ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_isis.py

```diff
@@ -74,12 +74,24 @@ def router_isis(self) -> dict | None:
     @cached_property
     def _isis_net(self) -> str | None:
         isis_system_id_prefix = get(self.shared_utils.switch_data_combined, "isis_system_id_prefix")
+
         if isis_system_id_prefix is not None:
             isis_area_id = get(self._hostvars, "isis_area_id", default="49.0001")
-            if self.shared_utils.id is None:
-                raise AristaAvdMissingVariableError(f"'id' is not set on '{self.shared_utils.hostname}' and is required to set ISIS NET address using prefix")
-
-            return f"{isis_area_id}.{isis_system_id_prefix}.{self.shared_utils.id:04d}.00"
+            isis_system_id_format = get(self._hostvars, "isis_system_id_format", default="node_id")
+            if isis_system_id_format in "node_id":
+                if self.shared_utils.id is None:
+                    raise AristaAvdMissingVariableError(
+                        f"'id' is not set on '{self.shared_utils.hostname}' and is required to set ISIS NET address using the node ID"
+                    )
+                net = f"{isis_area_id}.{isis_system_id_prefix}.{self.shared_utils.id:04d}.00"
+            else:
+                if self.shared_utils.router_id is None:
+                    raise AristaAvdMissingVariableError(
+                        f"'router_id' is not set on '{self.shared_utils.hostname}' and is required to set ISIS NET address using the underlay loopback"
```

Comment:
This is not needed. Router_id will *always* be available, and it will raise on it's own if any of the required things (like id and loopback_ipv4_pool) are missing.
```suggestion
```